app/src/main/java/com/android/feedme/ui/LandingScreen.kt

```diff
@@ -184,3 +191,66 @@ fun RecipeList(padding: PaddingValues, recipes: List<Recipe>) {
     }
   }
 }
+
+/**
+ * Function that displays our search bar as well as the filters icon
+ *
+ * @param searchText : the default text displayed in the search bar
+ * @param onSearchTextChanged : if we want to apply any changes to the text we type in the search
+ *   bar
+ * @param onFilterClicked : the function that will allow us to go through filters by clicking on the
+ *   icon
+ */
+@Composable
+fun SearchBar(
+    searchText: MutableState<String>,
+    onSearchTextChanged: (String) -> Unit,
+    onFilterClicked: () -> Unit
+) {
+  Row(
+      modifier = Modifier.fillMaxWidth().background(Color.White).padding(16.dp),
+      verticalAlignment = Alignment.CenterVertically) {
+        OutlinedTextField(
+            value = searchText.value,
+            onValueChange = {
+              searchText.value = it
+              onSearchTextChanged(it)
+            },
+            placeholder = { Text(text = "Find a friend or recipe", fontSize = 18.sp) },
+            leadingIcon = {
+              Icon(imageVector = Icons.Default.Search, contentDescription = "Search")
+            },
+            modifier = Modifier.weight(1f).background(Color.Transparent).testTag("SearchBar"),
+            shape = RoundedCornerShape(50) // Set a large rounded shape to create a pill effect
+            )
+        Spacer(modifier = Modifier.width(16.dp))
+        // The filters' icon
+        Icon(
+            imageVector = Icons.Default.FilterList,
+            contentDescription = "Filter",
+            modifier =
+                Modifier.size(48.dp)
+                    .clickable(onClick = onFilterClicked)
+                    .padding(8.dp)
+                    .testTag("FilterClick"))
+      }
+}
+
+/**
+ * Function responsible for the general landing page Contains namely the Search Bar and the Recipes
+ * List
+ *
+ * @param padding : the padding because this will be our main component
+ * @param recipes : the argument to our RecipeList() function
+ */
+@Composable
+fun LandingScreenComplete(padding: PaddingValues, recipes: List<Recipe>) {
```

Comment:
Nitpick: instead of PaddingValues, you should probably pass a modifier that you then add a testTag and background to?